src/modules/powerrename/lib/Helpers.cpp

```diff
@@ -169,56 +169,98 @@ HRESULT GetTransformedFileName(_Out_ PWSTR result, UINT cchMax, _In_ PCWSTR sour
     return hr;
 }
 
+bool isFileAttributesUsed(_In_ PCWSTR source) 
+{
+    bool used = false;
+    std::wstring patterns[] = { L"(([^\\$]|^)(\\$\\$)*)\\$Y", L"(([^\\$]|^)(\\$\\$)*)\\$M", L"(([^\\$]|^)(\\$\\$)*)\\$D", 
+        L"(([^\\$]|^)(\\$\\$)*)\\$h", L"(([^\\$]|^)(\\$\\$)*)\\$m", L"(([^\\$]|^)(\\$\\$)*)\\$s", L"(([^\\$]|^)(\\$\\$)*)\\$f" };
+    size_t patternsLength = ARRAYSIZE(patterns);
+    for (size_t i = 0; !used && i < patternsLength; i++)
+    {
+        if (std::regex_search(source, std::wregex(patterns[i])))
+        {
+            used = true;
+        }
+    }
+    return used;
+}
 HRESULT GetDatedFileName(_Out_ PWSTR result, UINT cchMax, _In_ PCWSTR source, SYSTEMTIME LocalTime)
 {
+    PWSTR months[] = { L"Jan", L"Feb", L"Mar", L"Apr", L"May", L"Jun", L"Jul", L"Aug", L"Sep", L"Oct", L"Nov", L"Dec"};
+    PWSTR monthsFullName[] = { L"January", L"February", L"March", L"April", L"May", L"June", L"July", L"August", L"September", L"October", L"November", L"December" };
+    PWSTR days[] = { L"Sun", L"Mon", L"Tue", L"Wed", L"Thu", L"Fri", L"Sat" };
+    PWSTR daysFullName[] = { L"Sunday", L"Monday", L"Tuesday", L"Wednesday", L"Thursday", L"Friday", L"Saturday" };
```

Comment:
These strings need to be localizable, check if there is an API to get the localized versioned from the system.